src/script.c

```diff
@@ -108,10 +108,62 @@ int scriptInterrupt(scriptRunCtx *run_ctx) {
 }
 
 /* Prepare the given run ctx for execution */
-void scriptPrepareForRun(scriptRunCtx *run_ctx, client *engine_client, client *caller, const char *funcname) {
+int scriptPrepareForRun(scriptRunCtx *run_ctx, client *engine_client, client *caller, const char *funcname, uint64_t function_flags, int ro) {
     serverAssert(!curr_run_ctx);
-    /* set the curr_run_ctx so we can use it to kill the script if needed */
-    curr_run_ctx = run_ctx;
+
+    if (!(function_flags & SCRIPT_FLAG_IGNORE_FLAGS)) {
+        if ((function_flags & SCRIPT_FLAG_NO_CLUSTER) && server.cluster_enabled) {
+            addReplyError(caller, "Can not run function on cluster, 'no-cluster' flag is set.");
+            return C_ERR;
+        }
+
+        if (!(function_flags & SCRIPT_FLAG_ALLOW_OOM) && server.script_oom && server.maxmemory) {
+            addReplyError(caller, "-OOM allow-oom flag is not set on the function, "
+                                  "can not run it when used memory > 'maxmemory'");
+            return C_ERR;
+        }
+
+        if (server.masterhost && server.repl_state != REPL_STATE_CONNECTED &&
+            server.repl_serve_stale_data == 0 && !(function_flags & SCRIPT_FLAG_ALLOW_STALE))
```

Comment:
If you make them stale you will break script that do not use the new shebang no? I guess you need check for stale on the eval/eval_ro callback only if shebang is given?

Reply:
I guess this will just work no? because if there is no shebang we will not performing those tests. but maybe, if there is no shebang, we should not verify stale on scriptCall (to not break): https://github.com/redis/redis/blob/55c81f2cd3da82f9f570000875e006b9046ddef3/src/script.c#L421?

Reply:
Changed `EVAL[SHA]/_RO` to support `STALE` flag. Added backward compatibility mode so it looks like they don't support `STALE` when there's no shebang. Please review.